libsplinter/src/auth/oauth/builder/mod.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Builders for [OauthClient](crate::auth::oauth::OauthClient) structs.
+
+#[cfg(feature = "oauth-github")]
+mod github;
+#[cfg(feature = "oauth-openid")]
+mod openid;
+
+use crate::auth::rest_api::identity::IdentityProvider;
+use crate::error::InvalidStateError;
+
+use super::error::OAuthClientBuildError;
+use super::OAuthClient;
+
+#[cfg(feature = "oauth-github")]
+pub use github::GithubOAuthClientBuilder;
+
+#[cfg(feature = "oauth-openid")]
+pub use openid::OpenIdOAuthClientBuilder;
+
+/// Builds a new `OAuthClient`
+///
+/// # Arguments
+///
+/// * `client_id` - The OAuth client ID
+/// * `client_secret` - The OAuth client secret
+/// * `auth_url` - The provider's authorization endpoint
+/// * `redirect_url` - The endpoint that the provider will redirect to after it has completed
+///   authorization
+/// * `token_url` - The provider's endpoint for exchanging an authorization code for an access
+///   token
+/// * `scopes` - The scopes that will be requested for each user
+/// * `identity_provider` - The OAuth identity provider used to retrieve the users' identity
+///
+/// # Errors
+///
+/// Returns an error if any of the auth, redirect, or token URLs are invalid
```

Comment:
This doc comment shouldn't include arguments or errors, and the description should probably be updated a bit. Same for the GitHub and OpenID builders.

Reply:
Fixed.